specification/columns/commitmentdiscountquantity.md

```diff
@@ -2,9 +2,11 @@
 
 Commitment Discount Quantity is the amount of a [*commitment discount*](#commitment-discount) purchased or accounted for in *commitment discount* related [*rows*](#glossary:row). The amounts are denominated in [*Commitment Discount Units*](#glossary:commitmentdiscountunit). For a purchase, the Commitment Discount Quantity is either the total of *Commitment Discount Units* over a *commitment discount's* [*term*](#glossary:term) or the apportioned amount for the [*charge period*](#glossary:chargeperiod). For committed usage, the Commitment Discount Quantity is either the number of *Commitment Discount Units* consumed by a [*row*](glossary:#row) that is covered by a *commitment discount* or is the unused portion of a *commitment discount* over a *charge period*. The CommitmentDiscountQuantity column only applies to *commitment discounts* and not [*negotiated discounts*](#glossary:negotiated-discount). Commitment Discount Quantity is commonly used in commitment discount analysis and optimization use cases.
 
+For usage-based *commitment discounts* (i.e. when *CommitmentDiscountCategory* is "Usage"), the Commitment Discount Quantity reflects the amount of resource hours consumed. If *size-flexibility* is applicable, this amount is further multiplied by a pre-determined, SKU attribute called an *instance type ratio*, to produce a normalized quantity based on [*resource's*](#glossary:resource) [*instance-size*](#glossary:instance-size). *Size-flexibility* allows a usage-based *commitment discount* to cover less expensive SKUs with smaller *instance type ratios*, or conversely, partially cover a more expensive SKU with a larger *instance type ratio*.
```

Comment:
Not following, that is not a ratio relative to the instance type but rather the size, normalization factor typically isn't instance type centric but rather size.

Reply:
This is also implying expense is the determining factor, however that is entirely up to the providers chosen method for applicability.

Reply:
Not sure if we should tie size flexibility to instances and hours or should we mention that as an example and leave room for other types of resources?

Additionally, we are introducing several terms into the glossary (e.g., Instance Family, Instance Size, Instance Type Ratio) that we need consensus on. Can we achieve that on such short notice? Should we be cautious about introducing new terms into the glossary, or is it acceptable to potentially rename or redefine them in later releases?

Since I'm really interested in **getting this PR in** and believe it's **important to mention size flexibility** (highlighting the difference between this and other quantity columns), **in case we need to proceed cautiously and don't have that consensus**, here is an extremely simplified alternative:

> _Unlike Pricing Quantity and Consumed Quantity, Commitment Discount Quantity includes commitment discount size flexibility._

**Note:** This would need to be described in more detail here or in the corresponding appendix, and maybe we can reintroduce this paragraph (or something similar to this) later once we agree on the terms, etc.

For reference, in the SkuPriceId column, we've used the term `commitment discount flexibility` and assumed that practitioners are familiar with it or would seek further information if needed.

> _A given value of SkuPriceId MUST be associated with one and only one SkuId, except in cases of commitment discount flexibility._

Reply:
At this point, I'd recommend getting the pr in with the most minimal mention of size flex possible and following up later.

Reply:
@ijurica I agree, however in this case there is both a usecase of numerically representing the quantity in terms of normalized size flexibility and non normalized terms. If for example I have two RI's for m5.4xlarge and deploy 2 m5.2xlarges. I would want to know in relation to the RIs purchase family and size how many went unused/used (2  in this case) and how many normalized units when used unused in terms of the base size flexible units (32 in this case).

Reply:
This is derivable by comparing the consumed quantity/unit of what was used against the used and unused commitmentdiscountquantity.